### PR TITLE
Fixing code smells for user login and registration use cases

### PR DESCRIPTION
**Pull request by daexs**
This pull requests fixes code smells and adds java doc to classes in the user login and registration use cases. Naming conventions were also fixed.

This pull request modified: 
- LoginUI
- Main
- RegisterUI
- TemporaryMemoryAccess
- UserLoginPresenter
- UserLoginInteractorTest
- UserRegisterInteractor Test
- UserLoginPresenter

### CODE REVIEW

File: src/main/java/screens/LoginUI.java [hunk 1]
```diff
@@ -35,62 +35,59 @@ public static void main(String[] args) {
 
     public void start(Stage primaryStage){
```
Comment (oscartuvey) [inline src/main/java/screens/LoginUI.java:36]: Maybe add Javadoc to this method?

Comment (oscartuvey) [reply src/main/java/screens/LoginUI.java:36]: A few comments within the method itself might make things clearer for someone reading the code.

File: src/main/java/screens/RegisterUI.java [hunk 2]
```diff
@@ -30,31 +38,31 @@ public void start(Stage primaryStage){
...
+        formGridPane.add(reconfirmPasswordLabel,0,2);
+        formGridPane.add(repeatedPasswordField,2,2);
 
         Label choose = new Label("Please choose a user type:");
         Label error = new Label();
-        Button regis = new Button("Register");
+        Button register = new Button("Register");
 
         Button goBackButton = new Button("Go back");
         goBackButton.getStyleClass().add("go-back-button");
```
Comment (oscartuvey) [inline src/main/java/screens/RegisterUI.java:68]: Perhaps add some Javadoc for the start method?

**Review by oscartuvey — CHANGES_REQUESTED** [review]
Perhaps add some Javadoc to a few of the methods I have annotated. Besides that though, well done for changing the name of all the variables to fit naming conventions!

File: src/main/java/screens/UserCreationError.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package screens;
+
+public class UserCreationError extends RuntimeException{
```
Comment (oscartuvey) [inline src/main/java/screens/UserCreationError.java:6]: Add Javadoc.

File: src/main/java/screens/UserLoginError.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package screens;
+
+public class UserLoginError extends RuntimeException{
```
Comment (oscartuvey) [inline src/main/java/screens/UserLoginError.java:6]: Add Javadoc.

**Review by oscartuvey — CHANGES_REQUESTED** [review]
Javadoc needed for the two new exceptions

File: src/main/java/screens/UserCreationError.java (hunk 3 above)
Comment (daexs) [reply src/main/java/screens/UserCreationError.java:6]: Added the javadoc

File: src/main/java/screens/UserLoginError.java (hunk 4 above)
Comment (daexs) [reply src/main/java/screens/UserLoginError.java:6]: Added the javadoc

**Review by oscartuvey — APPROVED** [review]
LGTM

**Pull request merged**